locations/spiders/audi.py

```diff
@@ -0,0 +1,131 @@
+        "BGR",
+        "BHR",
+        "BIH",
+        "BLR",
+        "BRA",
+        "BRN",
+        "CAN",
+        "CHE",
+        "CHL",
+        "CHN",
+        "CYP",
+        "CZE",
+        "DEU",
+        "DNK",
+        "ESP",
+        "EST",
+        "FIN",
+        "FRA",
+        "GBR",
+        "GEO",
+        "GRC",
+        "HKG",
+        "HRV",
+        "HUN",
+        "IDN",
+        "IND",
+        "IRL",
+        "ISL",
+        "ISR",
+        "ITA",
+        "JOR",
+        "JPN",
+        "KAZ",
+        "KOR",
+        "KWT",
+        "LBN",
+        "LTU",
+        "LUX",
+        "LVA",
+        "MDA",
+        "MEX",
+        "MKD",
+        "MLT",
+        "MNE",
+        "MYS",
+        "NLD",
+        "NOR",
+        "NZL",
+        "OMN",
+        "PHL",
+        "POL",
+        "PRT",
+        "QAT",
+        "ROU",
+        "RUS",
+        "SAU",
+        "SGP",
+        "SRB",
+        "SVK",
+        "SVN",
+        "SWE",
+        "THA",
+        "TUR",
+        "TWN",
+        "UKR",
+        "USA",
+        "UZB",
+        "ZAF",
+    ]
+
+    def start_requests(self):
+        for iso_3 in self.countries:
+            iso_2 = CountryUtils().to_iso_alpha2_country_code(country_str=iso_3)
+            for lat, lon in point_locations("eu_centroids_120km_radius_country.csv", iso_2):
+                yield JsonRequest(
+                    url=self.graphql_url,
+                    method="POST",
+                    data={
+                        "operationName": "Dealer",
+                        "variables": {"country": iso_3, "lat": float(lat), "lng": float(lon)},
+                        "query": "query Dealer($country: String, $services: String, $lng: Float!, $lat: Float!) {\n  dealersByGeoLocation(\n  limit: 100\n    country: $country\n    serviceFilter: $services\n  lat: $lat\n lng: $lng\n) {\n    dealers {\n      ...FragmentDealerFields\n      __typename\n    }\n    meta {\n      resultCount\n      __typename\n    }\n    __typename\n  }\n}\n\nfragment FragmentDealerFields on Dealer {\n  dealerId\n  vCardUrl\n matPrimaryCode\n  kvpsPartnerKey\n  name\n  address\n  latitude \n  longitude\n houseNumber\n  street\n region\n  city\n  zipCode\n  services\n  distance\n  phone\n  additionalData {\n    displayName\n    locationName\n    __typename\n  }\n  openingHours {\n    departments {\n      id\n      departmentName\n      openingHours {\n        id\n        open\n        timeRanges {\n          openTime\n           \n          __typename\n        }\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n  __typename\n}\n",
+                    },
+                    meta={"country_code": iso_2},
+                )
+
+    def parse(self, response):
+        data = response.json()
+        store_data = data.get("data").get("dealersByGeoLocation").get("dealers")
+        count = data.get("data").get("dealersByGeoLocation").get("meta").get("resultCount")
+        assert count < 100
```

Comment:
Is this meant to stay in?

Reply:
no, i removed it.
however how should we raise a warning if the number of pois is higher than 100 ?

Reply:
cause in this case, we will miss some

Reply:
Urm, `self.logger` would be the way, but I think count is the amount of results sent, so it was never going to be over 100, and I don't know if you are dropping POIs anyway as you are doing a request every 120km.

But the `dealersByMarket` returns 113 for "A-GBR" and 1,237 for "A-DEU" so it needs to be removed when switching to that query.